docs/design/add-in-commands.md

```diff
@@ -23,13 +24,20 @@ Add-in commands are UI elements that extend the Office UI and start actions in y
 
 ![Screenshot showing add-in commands in Excel on the web.](../images/add-in-commands-2.png)
 
+## Types of add-in commands
+
+There are two types of add-in commands, based on the kind of action that the command triggers:
+
+- **Task pane commands**: The button or menu item opens the add-in's task pane. You add this kind of add-in command with simple markup in the manifest. The "code behind" the command is provided by Office.
+- **Function commands**: The button or menu item runs any arbitrary JavaScript. The code almost always calls APIs in the Office JavaScript Library, but it doesn't have to. This type of add-in typically displays no UI other than the button or menu item itself. Note the following about function commands:
+
+   - Although they typically don't display a UI, the function that is triggered can call the [displayDialogAsync](/javascript/api/office/office.ui?view=common-js&preserve-view=true#office-office-ui-displaydialogasync-member(1)) method to show a dialog, which is a good way to display an error, show progress, or prompt for input from the user. If the add-in is configured to use a shared runtime, the function can also call the [showAsTaskpane](/javascript/api/office/office.addin#office-office-addin-showastaskpane-member(1)) method.
```

Comment:
This felt redundant after the previous sentence that says function commands don't usually display other UI components.
```suggestion
   - The function that is triggered can call the [displayDialogAsync](/javascript/api/office/office.ui?view=common-js&preserve-view=true#office-office-ui-displaydialogasync-member(1)) method to show a dialog, which is a good way to display an error, show progress, or prompt for input from the user. If the add-in is configured to use a shared runtime, the function can also call the [showAsTaskpane](/javascript/api/office/office.addin#office-office-addin-showastaskpane-member(1)) method.
```